src/components/@shared/UI/Input.tsx

```diff
@@ -0,0 +1,70 @@
+import { ReactNode } from "react";
+import { twMerge } from "tailwind-merge";
+
+interface InputProps {
+  type?: string; // 입력 필드 타입
+  placeholder?: string; // placeholder
+  buttonContent?: ReactNode; // 버튼에 들어갈 이지미, 텍스트
+  errorMessage?: string; // 에러 메시지
+  isValid?: boolean; // 유효성 검사
+  inputClassName?: string; // input 스타일
+  buttonClassName?: string; // 버튼 스타일
+  value?: string;
+  onChange?: (e: React.ChangeEvent<HTMLInputElement>) => void; // 입력 값 변경 핸들러
+  onButtonClick?: () => void; // 버튼 클릭 핸들러
+}
```

Comment:
type, placeholder 같은건 input에 그대로 내려주기 때문에 하나하나 정의하기보다 InputHTMLAttributes를 extends 하도록 해주는 게 좋아보입니다.
```suggestion
interface InputProps extends React.InputHTMLAttributes<HTMLInputElement> {
  buttonContent?: ReactNode; // 버튼에 들어갈 이지미, 텍스트
  errorMessage?: string; // 에러 메시지
  isValid?: boolean; // 유효성 검사
  buttonClassName?: string; // 버튼 스타일
  onButtonClick?: () => void; // 버튼 클릭 핸들러
}
```